aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2284,13 +2300,30 @@ fn create_account_if_does_not_exist(
         .map(|_return_vals| ())
 }
 
+/// Signals that the transaction should trigger the flow for creating an account as part of a
+/// sponsored transaction. This occurs when:
+/// * The feature gate is enabled SPONSORED_AUTOMATIC_ACCOUNT_V1_CREATION
+/// * There is fee payer
+/// * The sequence number is 0
+/// * There is no account resource for the account
 pub(crate) fn is_account_init_for_sponsored_transaction(
     txn_data: &TransactionMetadata,
     features: &Features,
-) -> bool {
-    txn_data.fee_payer.is_some()
-        && txn_data.sequence_number == 0
-        && features.is_enabled(FeatureFlag::SPONSORED_AUTOMATIC_ACCOUNT_V1_CREATION)
+    resolver: &impl AptosMoveResolver,
+) -> Result<bool, VMError> {
+    Ok(
+        features.is_enabled(FeatureFlag::SPONSORED_AUTOMATIC_ACCOUNT_V1_CREATION)
+            && txn_data.fee_payer.is_some()
+            && txn_data.sequence_number == 0
+            && resolver
+                .get_resource(&txn_data.sender(), &AccountResource::struct_tag())
```

Comment:
Not at the laptop, but don't we have exists_resource or similar method in the resolver?

(You can proceed with landing and cherry picking to 1.9 as is, maybe worth cleaning on main)

Reply:
`resource_exists_in_group` but that's more about the type and not the data, from what I recall.

Reply:
I was talking about `resource_exists` method defined here - https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/aptos-vm-types/src/resolver.rs#L54

`resource_exists_in_group` is for whether tag exists in resource group.

Reply:
ahh, neat!

Reply:
cool part about it, is that if transaction BlockSTM executes concurrently modifies the value (with resource existing before and after) - it will not be treated as a conflict. Where get_resource would need to re-execute - because it doesn't value is being ignored

Reply:
right, right, the only conflict might be if it is created or deleted, I imagine?

Reply:
yup :)